### ISSUE TITLE
Craft 4 support

### ISSUE THREAD

**Issue by turnstylerj** [jalendport/craft-fetch#12]
It looks like this plugin is not supported with Craft 4. Will there be an update?

**Comment by turnstylerj** [comment]
Including a link to some Discord conversation about this: https://discord.com/channels/456442477667418113/456442477667418115/971819344264847420

**Comment by patrikalienus** [comment]
> Including a link to some Discord conversation about this: https://discord.com/channels/456442477667418113/456442477667418115/971819344264847420

This requires login and I for one, do not use Discord.
If this convo is important to this issue, could you paste it here instead?

**Comment by gixxy22** [comment]
Any news on craft4 support? Dont use discord

**Comment by turnstylerj** [comment]
Attached a few screenshots of the chat for those who don't use Discord.

TLDR: He's thinking about it, Verbb dev sounded potentially interested in taking it on if he does ditch it. Would be curious if @jalendport has thought about it since.

![Screen Shot 2022-06-20 at 11 40 06 PM](https://user-images.githubusercontent.com/42223501/174733555-453b5abe-1590-49e4-93eb-9dc75cf074b7.png)
![Screen Shot 2022-06-20 at 11 40 40 PM](https://user-images.githubusercontent.com/42223501/174733571-959e12f6-fe41-4fc5-98ce-ad3644b73172.png)
![Screen Shot 2022-06-20 at 11 41 00 PM](https://user-images.githubusercontent.com/42223501/174733573-60ee0cb9-5606-4bda-a7d0-37be72b95d74.png)

**Comment by engram-design** [comment]
Just to chime in, we've added this to our [Craft 3 migration project](https://verbb.io/craft-3).

If you're looking for a new home for Fetch, we'd be happy to continue development at Verbb (including updating to Craft 4) - but just let us know if you've got plans to continue things and update to Craft 4! @jalendport

**Comment by engram-design** [comment]
@jalendport Just checking in to see what you're planning?

**Comment by splendidrob** [comment]
+1 for Craft 4!

**Comment by patrikalienus** [comment]
> Just to chime in, we've added this to our [Craft 3 migration project](https://verbb.io/craft-3).
> 
> If you're looking for a new home for Fetch, we'd be happy to continue development at Verbb (including updating to Craft 4) - but just let us know if you've got plans to continue things and update to Craft 4! @jalendport

I don't see it on the list of plugins on that page. Was it removed?

**Comment by engram-design** [comment]
@patrikalienus I have left this with @jalendport because he's told me that he'll be porting his plugins to Craft 4 himself. As such, you're correct that I've removed it from our list.

**Comment by turnstylerj** [comment]
Any possibility of a status update on this @jalendport even if just whether an update is still planned at some point? We have a few projects using it, and it'd be good to know if we should proceed under the assumption that this plugin is abandoned as we do Craft 4 upgrades.

**Comment by jannisborgers** [comment]
+1 for a Craft 4 port.

We tried a couple of methods to fetch an external GraphQL query and, but none of them worked correctly (with most, we couldn’t even get a valid response back). Only Fetch made it possible without the added work messing around with PHP modules etc. If someone can point me to a resource where this kind of thing is documented, I’d be happy to read it.

We’re now at the point where we would really like to upgrade the project to Craft 4. Mainly for features in plugins which are only available in their Craft 4 versions. We cannot upgrade though, because a whole site in the installation requires Fetch to work.

PS: The page speed is fine in our case. Plus we’re serving server side, crawlable pages without having to rely on the usual JS stuff, which is really all we need for this project.

**Comment by bartrylant** [comment]
For people still waiting after almost a year, you can now get the Verbb Consume plugin which does the same and more. I just used it on a big website that was waiting to be updated to Craft 4.

**Comment by jalendport** [comment]
Here's an update on where things are at: https://twitter.com/jalendport/status/1628813156888809475

In the meantime, @bartrylant's suggestion is a good one if you need a Fetch alternative immediately 👍

**Comment by jannisborgers** [comment]
Thanks Jalen for the update. Hope you slowly get back to normality, that thread read like a true horror story ;)

I’ve had the Verbb Consume tab open on my phone for weeks, but will wait some more time. I would appreciate it if you would tell us once you make a decision regarding Fetch (update or abandon), though.

**Comment by jalendport** [comment]
Just released [2.0.0-beta.1](https://github.com/jalendport/craft-fetch/releases/tag/2.0.0-beta.1), which is a fairly straightforward upgrade of the codebase to be Craft 4 ready. It's tagged as a beta at this point because I haven't had a chance to test it very thoroughly yet, and also because I'd like to implement/close some of the issues and PRs before I tag a final version of 2.0.0.

Thanks for y'all's patience - let me know if you run into issues with the new version and I'll make sure they're fixed right quick 💪
